dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AppController.java

```diff
@@ -273,11 +276,28 @@ public void renderApp(
         response.setContentType(mimeType);
       }
 
-      response.setContentLengthLong(resource.contentLength());
-      response.setHeader(
-          "Last-Modified", DateUtils.toHttpDateString(new Date(resource.lastModified())));
-
-      StreamUtils.copyThenCloseInputStream(resource.getInputStream(), response.getOutputStream());
+      if (filename.endsWith(".html")) {
```

Comment:
Can this condition be made more strict? 
I would assume we know more about which HTML documents need the below line-by-line replacement or is it really any of them?

Reply:
As discussed in today's meeting: we can limit this to a fixed list of files.